rest_framework/utils/mediatypes.py

```diff
@@ -46,7 +45,7 @@ def order_by_precedence(media_type_lst):
 class _MediaType:
     def __init__(self, media_type_str):
         self.orig = '' if (media_type_str is None) else media_type_str
-        self.full_type, self.params = parse_header(self.orig.encode(HTTP_HEADER_ENCODING))
+        self.full_type, self.params = parse_header_params(self.orig, encoding=HTTP_HEADER_ENCODING)
```

Comment:
This looks at first like it could be an issue, but I checked -- the class `_MediaType` is really used only in two places (`versioning.py` and `negotiation.py`), both of them in very localized ways; in both cases, values taken out of the dict immediately get decoded to `str`.